### ISSUE TITLE
Add Support for Multiple QR-based hardware wallets

### ISSUE THREAD

**Issue by shekohex** [RabbyHub/Rabby#1936]
Currently, I can't add two or more QR-based hardware wallets to Rabby. And even I can't add more than one of two different wallets like (Airgap and Keystone).

Related issues: #1314 and #557

Use case:
Currently, I have a few old phones and on each one of them I do have airgap vault installed, and basically I want to use them as a multisig wallet with Safe (https://safe.global).

P.S I believe Metamask supports adding more than one Keystone/QR Based wallets.

**Comment by vvvvvv1vvvvvv** [comment]
MetaMask is not support adding more than one QR Based wallets either, we have discussed with Keystone team about this but unfortunately can't resolve this problem currently

**Comment by shekohex** [comment]
> MetaMask is not support adding more than one QR Based wallets either, we have discussed with Keystone team about this but unfortunately can't resolve this problem currently

Yeah, I did try Metamask too, and it looks like it is an issue with the protocol? Will check Keystone/Airgap for that.

**Comment by shekohex** [comment]
For Airgap, why not use their [official Protocol](https://github.com/airgap-it/airgap-coin-lib) instead of using Keystone's one? In Airgap wallet, I can add many Airgap vaults to one wallet, and it works this way without any issues.

**Comment by suicide** [comment]
I spent some time looking into the wallet and keyring service code and I cannot see any hard stoppers that would prevent you from having multiple QR keyrings apart  from some of the wallet code being written to only support one keyring of a certain type.

@vvvvvv1vvvvvv could you give us a hint on where the issue lies and how keystone is involved?

also a bit off topic: is there a dev chat or something where one can ask questions about or discuss the code? There are some parts of the code that I do not understand or why they are necessary.

**Comment by MAHDTech** [comment]
Does this apply to using Keystone with USB?

Can you add 2x Keystone with USB? (I'm trying to but getting the warning below)

> It's not supported to import multiple Keystone devices If you switch to a new Keystone device, the current device's address list will be removed before starting the import process.

**Comment by krogla** [comment]
+1 feature request,
the same issue when using several seed phrases on one keystone

**Comment by Savid** [comment]
Would love multiple keystone device support as well as multiple account on a single keystone support!

**Comment by teledamage** [comment]
+1
this is an issue for me too, having to remove and reconnect a new wallet is not best practice imo.
there is one extension out there that supports multiple QR keystone wallets. (okx)
even better would be support via USB too, im guessing thats more challenging to implement though.

**Comment by 3Galley** [comment]
+1 super requesting this feature! 
Its a must have with Keystone 3 Pro. The device currently supports 3 separate seeds but I have to reset Rabby every time im switching to another one in the same device. 

Both via USB and via QR code Rabby says only takes one wallet at any given time on the Keystone 3 Pro. I get the same response from both the Chrome extension and the Android App. 

Would be amazing not having to reset Rabby every time im switching wallets. 

Please add this feature :)
Is there a way to create like a bounty ? to help contribute towards this feature

**Comment by 3Galley** [comment]
> [@3Galley](https://github.com/3Galley) instead of resetting you can add browser profiles and add the keystone to a new rabby inside those and switch like that, it also works well with USB having all three set up and turning on/off to access the accounts.
> 
> That being said, I would still really prefer having the ability to add them all in one rabby instance. please add this feature!

100%, thats one workaround, but Rabby already supports multiple wallets with the same physical Ledger hardware wallet. Would be awesome if the same functionality can be integrated to Keystone via QR and via USB under the same instance :)
